openedx/core/djangoapps/waffle_utils/tests/test_views.py

```diff
@@ -144,11 +144,26 @@ def test_code_owners_without_module_information(self):
         ][0]
         self.assertNotIn("code_owner", result)
 
+    def test_course_overrides(self):
+        models.WaffleFlagCourseOverrideModel.objects.create(waffle_flag="my.flag", enabled=True)
+        course_overrides = {}
+
+        # pylint: disable=protected-access
+        toggle_state_views._add_waffle_flag_course_override_state(course_overrides)
+        toggle_state_views._add_waffle_flag_computed_status(course_overrides)
```

Comment:
I find this PR a little strange, but maybe just because I don't get your game plan. There ultimately needs to be some way for the new version of this view in edx-toggles to include data from the CourseWaffleFlags that will remain in edx-platform.

My game plan would have been:
* Create the new view in edx-toggles with the capability to gather additional data (i.e. from CourseWaffleFlags). I'm unclear what shape this will take.
* Add whatever edx-platform code is required to add the CourseWaffleFlag data to this view's output.
* Switch to the new view and delete the code from the old view.

In this way, any refactors done would clearly be necessary for the final design.  In this PR, you are refactoring and testing functions like `_add_waffle_flag_course_override_state` that may take a different shape in the final design, so why spend time on this version of the code?

Given this, I don't see anything concerning here, so if you think it will make things easier for you to get to the final goal, then you can move forward with this.  I'd just like to see you more quickly jump to the edx-toggles version with the enhanced design.

Also, I hope the final tests (at least those in edx-toggles) can avoid testing the private methods and can instead test the view itself.

Reply:
I must admit that I did not give much thought to the migration of the view to another package; here I'm just trying to address the "Improve test coverage" bullet item from the [BD-21 page](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1596358943/BD-21+Toggles+Settings+Documentation). 

It was not clear to me that the toggle state view should be moved to edx-toggles. Is this a decision that we already made? It would require edx-toggles to depend on `rest_framework` and `edx_rest_framework`, which I don't think is desirable.

Also, I did not realize either that CourseWaffleFlag data would have to be added to this view.

Reply:
1. The migration was documented in the original ADR: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/waffle_utils/docs/decisions/0002-waffle-utils-extraction.rst
2. The endpoint is needed for other IDAs, including for use in the Toggle State Report spreadsheet that uses this data.  The repo edx-toggles seems like the obvious location, even though it adds new dependencies, but I am open to alternative proposals.

Reply:
Also, I'm pretty sure we are on the same page about CourseWaffleFlag, but I am just confirming that it will **not** be extracted from edx-platform. See https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/waffle_utils/docs/decisions/0003-leave-course-waffle-flag.rst

Reply:
> The endpoint is needed for other IDAs, including for use in the Toggle State Report spreadsheet that uses this data. The repo edx-toggles seems like the obvious location, even though it adds new dependencies, but I am open to alternative proposals.

I think that the best way to proceed is to expose a `get_all_toggles()` method in edx-toggles:

```
def get_all_toggles():
    response = OrderedDict()
    response['waffle_flags'] = self._get_all_waffle_flags()
    response['waffle_switches'] = self._get_all_waffle_switches()
    response['django_settings'] = self._get_settings_state()
    return response
```

It would then be the responsibility of every IDA to properly expose the response of this function, with the right permissions etc. This then makes it easier for other apps to add extra data, such as CourseWaffleFlag objects for instance. That way, edx-toggles does not need extra requirements.

What do you think?

> Also, I'm pretty sure we are on the same page about CourseWaffleFlag, but I am just confirming that it will not be extracted from edx-platform.

Indeed.

Reply:
> Most IDAs will just want the endpoint as-is.

The crux is in the "most"! The amount of extra code necessary to write a view (with auth, bells and whistles) is minimal, so I think it's better to keep it out of edx-toggles.

Reply:
@regisb: Sounds good. Please add a todo to implement in credentials, replacing this mock: https://github.com/edx/credentials/blob/fe2dade307c360f6d42b9449e8bb7ee449464e30/credentials/views.py#L21  Thank you!